launcher/cog-launcher.c

```diff
@@ -230,21 +230,33 @@ on_web_view_create(WebKitWebView *web_view, WebKitNavigationAction *action)
 static WebKitWebView *
 on_automation_session_create_web_view(WebKitAutomationSession *session, CogLauncher *launcher)
 {
+    static bool first_time = true;
+    if (first_time) {
+        first_time = false;
+        return (WebKitWebView *) cog_viewport_get_visible_view(launcher->viewport);
+    }
 #if HAVE_WEBKIT_AUTOPLAY
     g_autoptr(WebKitWebsitePolicies) website_policies =
         webkit_website_policies_new_with_policies("autoplay", s_options.autoplay_policy, NULL);
 #endif /* HAVE_WEBKIT_AUTOPLAY */
 
-    return (WebKitWebView *) cog_view_new("settings", cog_shell_get_web_settings(launcher->shell), "web-context",
-                                          cog_shell_get_web_context(launcher->shell), "is-controlled-by-automation",
-                                          TRUE, "zoom-level", s_options.scale_factor, "use-key-bindings", FALSE,
+    WebKitWebView *new_view =
```

Comment:
You could use `g_autoptr()` here and then we do not need the `g_object_unref()` below:

```suggestion
    g_autoptr(WebKitWebView) new_view =
```